src/searchee.ts

```diff
@@ -360,3 +394,251 @@ export function getReleaseGroup(stem: string): string | null {
 		? null
 		: parsedGroupMatchString;
 }
+
+const logReason = (reason: string, options: { useFilters: boolean }): void => {
+	if (!options.useFilters) return;
+	logger.verbose({
+		label: Label.PREFILTER,
+		message: reason,
+	});
+};
+
+function parseEnsembleKeys(
+	searchee: SearcheeWithLabel,
+	keys: string[],
+	ensembleTitles: string[],
+	episode: number | string,
+	existingSeasonMap: Map<string, SearcheeWithLabel[]>,
+	keyMap: Map<string, Map<number | string, SearcheeWithLabel[]>>,
+	ensembleTitleMap: Map<string, string>,
+): void {
+	for (let i = 0; i < keys.length; i++) {
+		const key = keys[i];
+		if (existingSeasonMap.has(key)) continue;
+		if (!ensembleTitleMap.has(key)) {
+			ensembleTitleMap.set(key, ensembleTitles[i]);
```

Comment:
I noticed the same obfuscated mutation here, although it looks like this one's entire purpose is to share the mutation code across anime/standard so maybe it's less avoidable. Renaming it might help

Reply:
There is a lot of mutation in this pipeline. I find it difficult to organize without it as the return types and reassignments would be a lot.